projects/cardiovascular-patient-app/src/app/report-bp/report-bp.component.ts

```diff
@@ -31,9 +40,28 @@ export class ReportBPComponent implements OnInit {
 
   ngOnInit(): void {
     this.form.valueChanges.subscribe((value) => {
-      console.log(this.layerManager.availableLayers$);
       this.updateBPentryForm(value);
     });
+    this.layerManager.allLayers$
+      .pipe(
+        map((layers) =>
+          layers
+            .filter((layer) => layer.name === 'Blood Pressure')
+            .map((layer) => layer.datasets.map((data) => data.data))
+            .map((layer) => layer.map((data) => data.slice(-1)))
+        )
+      )
+      .subscribe((data: any) => {
+        if (data.length > 0) {
+          this.isLastPriorBp = true;
+          this.lastReportedBPdata = {
+            systolic: { date: `${formatDate(data[0][1][0].x)} at ${formatTime(data[0][1][0].x)}`, value: data[0][1][0].y },
+            diastolic: { date: `${formatDate(data[0][0][0].x)} at ${formatTime(data[0][0][0].x)}`, value: data[0][0][0].y },
+          };
```

Comment:
This is confusing to call a layers array `data`. I mistook it for the data property of dataset at first and I was confused about why it is a 3-dimensional array. If you have an array of layers, call it `layers`.